src/unix/signal.c

```diff
@@ -234,20 +238,22 @@ static int uv__signal_register_handler(int signum, int oneshot) {
   if (oneshot)
     sa.sa_flags |= SA_RESETHAND;
 
-  /* XXX save old action so we can restore it later on? */
-  if (sigaction(signum, &sa, NULL))
+  if (sigaction(signum, &sa, &sa_old))
     return UV__ERR(errno);
 
+  uv__signal_handlers[signum] = sa_old.sa_handler;
```

Comment:
This won't work well. You need to store the full `struct sigaction`, otherwise you lose info like sa_flags. As well, it's technically possible for sa_handler and sa_sigaction to be distinct fields.

Reply:
@bnoordhuis Thank you for a comment. I'v pushed updates.

Reply:
@bnoordhuis Is everything ok now?